presto-cassandra/src/test/java/io/prestosql/plugin/cassandra/TestCassandraConnector.java

```diff
@@ -213,7 +216,7 @@ private static void assertReadFields(RecordCursor cursor, List<ColumnMetadata> s
                 if (BOOLEAN.equals(type)) {
                     cursor.getBoolean(columnIndex);
                 }
-                else if (INTEGER.equals(type)) {
+                else if (TINYINT.equals(type) || SMALLINT.equals(type) || INTEGER.equals(type)) {
```

Comment:
split this into 3 separate cases and apply validation on the `cursor.getLong` result using:

- `com.google.common.primitives.SignedBytes#checkedCast`
- `com.google.common.primitives.Shorts#checkedCast`
- `java.lang.Math#toIntExact`